arbnode/inbox_reader.go

```diff
@@ -159,6 +159,46 @@ func (r *InboxReader) Start(ctxIn context.Context) error {
 	return nil
 }
 
+// assumes l1block is recent so we could do a simple-search from the end
+func (r *InboxReader) recentL1BlockToMsg(ctx context.Context, l1block uint64) (arbutil.MessageIndex, error) {
```

Comment:
I think we should strongly consider adding an LRU cache here or somewhere nearby so people can't spam our RPC with finalized or safe requests, given that this involves a linear (albeit limited) scan.

Reply:
Ah I forgot there was an LRU cache there. Makes sense!